src/RevProc.cc

```diff
@@ -1635,22 +1670,25 @@ void RevProc::DependencySet(uint16_t HartID, uint16_t RegNum,
   }
 }
 
-uint16_t RevProc::GetHartID()const{
-  if(HART_CTS.none()) { return HartToDecode;};
+uint16_t RevProc::GetHartID() const {
+  // TODO: (BEFORE MERGE) --- Potentially make this a bitset that is MAX HARTS long
+  if( std::none_of(HART_CTS.begin(), HART_CTS.end(), [](bool x){ return true; } )){
+    return HartToDecode;
+  }
 
   uint16_t nextID = HartToDecode;
   if(HART_CTS[HartToDecode]){
     nextID = HartToDecode;
   }else{
-    for(int tID = 0; tID < _REV_HART_COUNT_; tID++){
+    for(size_t tID = 0; tID < Harts.size(); tID++){
       nextID++;
-      if(nextID >= _REV_HART_COUNT_){
+      if(nextID >= Harts.size()){
```

Comment:
I hated this function. In my initial rework, it printed out messages every time, so I just backed out my changes. I think there is a better more elegant way to handle it.

Reply:
Yeah, there's definitely a better way to do it. If I remember in your changes there was a scope that got muddied so you were seeing the message printed every time even if the hart didn't switch (or something similar) 

I'm also not crazy about the name because `GetHartID` implies you're getting the current Hart's ID when in reality you're getting the next Hart that is ready to decode. 

I think we can potentially modify the logic now that we have a RevHart object which can potentially allow to us to remove HART_CTE & HART_CTS as separate data structures in the future and perhaps cycle through each RevHart and check if its ready, then stop once you find one, but for now I'd say just deal with it.